### PR TITLE
mp-b2b-product-offer-prices-endpoint

### PR DESCRIPTION
**Pull request by himanshu-pal-metacube**
CC-16362   mp-b2b-product-offer-prices

### CODE REVIEW

File: tests/api/mp_b2b/glue/general_product_endpoints/product_offer_prices/positive.robot [hunk 1]
```diff
@@ -0,0 +1,18 @@
+*** Settings ***
+Resource    ../../../../../../resources/common/common_api.robot
+Suite Setup    SuiteSetup
+Test Setup    TestSetup
+Default Tags    glue
+
+*** Test Cases ***
+ENABLER
+    TestSetup
+Retrieve_prices_of_a_product_offer
```
Comment (OKrivtsova) [inline tests/api/mp_b2b/glue/general_product_endpoints/product_offer_prices/positive.robot:10]: please add tests for other price types (you can have a look at this PR to check what prices there are ad currencies, etc.) https://github.com/spryker/robotframework-suite-tests/pull/146/files
volume prices, original price, chf currency

File: tests/api/mp_b2b/glue/general_product_endpoints/product_offer_prices/negative.robot [hunk 2]
```diff
@@ -0,0 +1,24 @@
+*** Settings ***
+Resource    ../../../../../../resources/common/common_api.robot
+Suite Setup    SuiteSetup
+Test Setup    TestSetup
+Default Tags    glue
+
+*** Test Cases ***
```
[the remark this answers is not in the document]
Comment (himanshu-pal-metacube) [reply tests/api/mp_b2b/glue/general_product_endpoints/product_offer_prices/negative.robot]: Fixed

File: tests/api/mp_b2b/glue/general_product_endpoints/product_offer_prices/positive.robot (hunk 1 above)
Comment (himanshu-pal-metacube) [reply tests/api/mp_b2b/glue/general_product_endpoints/product_offer_prices/positive.robot]: Added

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    And Response should contain the array of a certain size:    [data][0][attributes][prices][0][volumePrices]    3
+    And Each array element of array in response should contain nested property with datatype:    [data]    [attributes][prices][0][volumePrices][0][grossAmount]    int
+    And Each array element of array in response should contain nested property with datatype:    [data]    [attributes][prices][0][volumePrices][0][netAmount]    int
+    And Each array element of array in response should contain nested property with datatype:    [data]    [attributes][prices][0][volumePrices][0][quantity]    int
+    And Response should contain the array of a certain size:    [data][0][attributes][prices][0][currency]    3
+    And Response body parameter should be:    [data][0][attributes][prices][0][currency][code]    ${currency_code_eur}
+    And Response body parameter should be:    [data][0][attributes][prices][0][currency][name]    ${currency_name_eur}
+    And Response body parameter should be:    [data][0][attributes][prices][0][currency][symbol]    ${currency_symbol_eur}
+
+Get_product_offer_price_with_gross_chf_volume_prices
```
Comment (OKrivtsova) [inline tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot:111]: volume prices are not actually checked here

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    And Response body parameter should be:    [data][0][id]    ${offer_with_volume_price}
+    And Response body parameter should be:    [data][0][type]    product-offer-prices
+    And Response body parameter should be:    [data][0][attributes][prices][0][netAmount]    None
+    And Response body parameter should be greater than:    [data][0][attributes][prices][0][grossAmount]    1
+    And Response should contain the array of a certain size:    [data][0][attributes][prices][0][currency]    3
+    And Response body parameter should be:    [data][0][attributes][prices][0][currency][code]    ${currency_code_chf}
+    And Response body parameter should be:    [data][0][attributes][prices][0][currency][name]    ${currency_name_chf}
+    And Response body parameter should be:    [data][0][attributes][prices][0][currency][symbol]    ${currency_symbol_chf}
+
+Get_product_offer_price_with_net_chf_volume_prices
```
Comment (OKrivtsova) [inline tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot:125]: volume prices are not actually checked here

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot (hunk 3 above)
Comment (himanshu-pal-metacube) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot:111]: Fixed

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot (hunk 4 above)
Comment (himanshu-pal-metacube) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot:126]: Fixed

File: tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot (hunk 3 above)
Comment (OKrivtsova) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot:111]: the fix just means that there are NO volume prices for this offer. what does the test then test? please switch to an offer that has volume prices. If there are no such offers remove the tests

Comment (himanshu-pal-metacube) [reply tests/api/mp_b2b/glue/product_offer_endpoints/product_offer_prices/positive.robot]: their is no offers with volume prices in CHF.

**Review by OKrivtsova — DISMISSED** [review]

**Review by OKrivtsova — APPROVED** [review]

**Pull request merged**